marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -548,17 +641,37 @@ handleGistAction settings PublishGist = do
                 Just gistId -> runAjax $ flip runReaderT settings $ Server.patchApiGistsByGistId newGist gistId
         assign _createGistResult newResult
         gistId <- hoistMaybe $ preview (_Success <<< gistId) newResult
-        assign _gistId (Just gistId)
-        assign _loadGistResult (Right NotAsked)
+        -- Mark all editors as saved
+        lift do
+          toHaskellEditor $ HaskellEditor.handleAction settings $ HE.MarkProjectAsSaved
+          toJavascriptEditor $ JavascriptEditor.handleAction settings $ JS.MarkProjectAsSaved
+          toSimulation $ Simulation.handleAction settings $ ST.MarkProjectAsSaved
+          void $ query _blocklySlot unit (Blockly.MarkProjectAsSaved unit)
+          void $ query _actusBlocklySlot unit (ActusBlockly.MarkProjectAsSaved unit)
+        modify_
+          ( set _gistId (Just gistId)
+              <<< set _loadGistResult (Right NotAsked)
+              {- This marks the project as saved globally, it would normally be a replication
+               of the inner unsaved state set below, but we n two places. Here to update the view -}
+              
+              <<< set _hasUnsavedChanges false
+          )
 
 handleGistAction _ (SetGistUrl url) = do
   case Gists.parseGistUrl url of
-    Right newGistUrl -> do
-      assign _createGistResult NotAsked
-      assign _loadGistResult (Right NotAsked)
-      assign _gistId (Just newGistUrl)
+    Right newGistUrl ->
+      modify_
+        ( set _createGistResult NotAsked
+            <<< set _loadGistResult (Right NotAsked)
+            <<< set _gistId (Just newGistUrl)
+        )
     Left _ -> pure unit
 
+-- TODO: I think this action is not being called.
```

Comment:
This is something that _definitely_ warrants a compiler warning.

Reply:
I added the warning but kept the TODO comment as well... The TODO is on line 670, where the handler is actually defined, the warning I added in line 611, where the signature of `handleGistAction` lives, and for some reason the compiler throws the warning in line 775, at the definition of `withAccidentalNavigationGuard` 🤷‍♂️